sim/pkg/simulator/heterogenous_stock.go

```diff
@@ -0,0 +1,69 @@
+package simulator
+
+import "fmt"
+
+type heterogenousStock struct {
+	name       StockName
+	stocksKind EntityKind
+	stock      map[Entity]bool
+}
+
+func (hs *heterogenousStock) Name() StockName {
+	return hs.name
+}
+
+func (hs *heterogenousStock) KindStocked() EntityKind {
+	return hs.stocksKind
+}
+
+func (hs *heterogenousStock) Count() uint64 {
+	return uint64(len(hs.stock))
+}
+
+func (hs *heterogenousStock) EntitiesInStock() []*Entity {
+	entities := make([]*Entity, 0)
+
+	for entity := range hs.stock {
+		entities = append(entities, &entity)
+	}
+	return entities
+}
+
+func (hs *heterogenousStock) Add(entity Entity) error {
```

Comment:
You might want to add a comment about the runtime complexity of Add / Remove O(1) versus EntitiesInStock O(n).  After all, that's the point of having two different base stocks.

Reply:
Added